sns/parity_math.c

```diff
@@ -541,6 +572,77 @@ static void xor_calculate(struct m0_parity_math *math,
 
 }
 
+#if ISAL_ENCODE_ENABLED
+static void isal_diff(struct m0_parity_math *math,
+		      struct m0_buf         *old,
+		      struct m0_buf         *new,
+		      struct m0_buf         *parity,
+		      uint32_t               index)
+{
+	uint8_t	 *diff_data = NULL;
+	uint32_t  block_size;
+	uint32_t  alignment = sizeof(uint_fast32_t);
+	uint32_t  i;
+	int	  ret = 0;
+
+	M0_ENTRY();
+
+	M0_PRE(parity_math_invariant(math));
+	M0_PRE(old    != NULL);
+	M0_PRE(new    != NULL);
+	M0_PRE(parity != NULL);
+	M0_PRE(index  <  math->pmi_data_count);
+	M0_PRE(old[index].b_nob == new[index].b_nob);
+
+	uint8_t	 *parity_frags[math->pmi_parity_count];
+
+	block_size = new[index].b_nob;
+
+	/* It is assumed that the buffer size will always be a multiple of
+	 * 8-bytes (especially since block size currently is 4K) and this assert
+	 * is added with the hope that any deviation from this assumption is
+	 * caught during development instead of on the field.
+	 */
+	M0_ASSERT_INFO(m0_is_aligned(block_size, alignment),
+		       "block_size=%u is not %u-bytes aligned",
+		       block_size, alignment);
+
+	ALLOC_ARR_INFO(diff_data, block_size, "differential data block", ret);
+	if (diff_data == NULL)
```

Comment:
This might not work since you would have asserted in the above macro ALLOC_ARR_INFO. See if you can avoid using this macro.

Reply:
Sorry my mistake. Ignore this comment.